src/app/screens/LNURLRedeem/index.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  CaretLeftIcon,
+  CrossIcon,
+  QrCodeIcon,
+} from "@bitcoin-design/bitcoin-icons-react/filled";
+import Button from "@components/Button";
+import Container from "@components/Container";
+import Header from "@components/Header";
+import IconButton from "@components/IconButton";
+import QrcodeScanner from "@components/QrcodeScanner";
+import TextField from "@components/form/TextField";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import lnurlLib from "~/common/lib/lnurl";
+import { isLNURLDetailsError } from "~/common/utils/typeHelpers";
+
+function LNURLRedeem() {
+  const { t } = useTranslation("translation", { keyPrefix: "lnurlredeem" });
+
+  const [invoice, setInvoice] = useState("");
+  const navigate = useNavigate();
+  const [qrIsOpen, setQrIsOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    try {
+      setLoading(true);
+
+      const lnurl = lnurlLib.findLnurl(invoice);
+
+      if (lnurl) {
+        const lnurlDetails = await lnurlLib.getDetails(lnurl);
+
+        if (isLNURLDetailsError(lnurlDetails)) {
+          toast.error(lnurlDetails.reason);
+          return;
+        }
+
+        if (lnurlDetails.tag === "withdrawRequest") {
+          navigate("/lnurlWithdraw", {
+            state: {
+              args: {
+                lnurlDetails,
+              },
+            },
+          });
+        } else {
+          toast.error("Not a lnurl-withdraw request");
+          return;
+        }
+      } else {
+        toast.error("Not a valid LNURL");
+        return;
+      }
+    } catch (e) {
+      if (e instanceof Error) {
+        toast.error(e.message);
+      }
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  function extractInvoiceFrom(data: string) {
+    const reqExp = /lightning=([^&|\b]+)/i;
+
+    const invoice = data.match(reqExp);
+
+    if (invoice) {
+      return invoice[1];
+    } else {
+      return data.replace(/^lightning:/i, "");
+    }
+  }
+
+  if (qrIsOpen) {
```

Comment:
I think a lot of this code is copied from src/app/screens/Send/index.tsx ?

I think it would be good to extract it into a single place if possible.

Reply:
tried to extract the scanner in at single place but we do set a state variable here. we need to return back a lnurl-withdraw code back. not sure if we shall go this way

Reply:
Can we have a screen component for the QR code which includes this?
![image](https://github.com/getAlby/lightning-browser-extension/assets/33993199/b53f2ac2-4fa3-440b-b3a3-7aaae76ce9fd)

Reply:
This could be moved into a ScanQRAdornment component with a callback function for qrCodeSuccessCallback and qrCodeErrorCallback. Would that work?

![image](https://github.com/getAlby/lightning-browser-extension/assets/33993199/597708ce-2059-4f88-b152-604db9127d7e)

Reply:
we need to consider here passing value back and getting back on the lnurl-redeem screen as well. if scanner extracted into separate screen. how callbacks will work here. in terms of passing value back and navigating?

currently i proposed a solution using navigation. instead of callbacks kindly have a look